query-engine/query-engine/src/server.rs

```diff
@@ -111,6 +131,17 @@ impl HttpServer {
         let mut res = match (req.method(), req.uri().path()) {
             (&Method::POST, "/") => {
                 let (parts, body) = req.into_parts();
+                let headers: HashMap<_, _> = parts
+                    .headers
+                    .iter()
+                    .map(|(k, v)| (format!("{}", k), v.to_str().unwrap().into()))
```

Comment:
trivia, nothing that should be fixed, but i tend to find `k.to_string()` easier to type than `format!("{}", k)` (they both use `Display` and do the same thing).